Source/WebCore/inspector/InspectorFrontendHost.cpp

```diff
@@ -636,6 +640,17 @@ void InspectorFrontendHost::setAllowsInspectingInspector(bool allow)
         m_client->setInspectorPageDeveloperExtrasEnabled(m_frontendPage->settings().developerExtrasEnabled());
 }
 
+bool InspectorFrontendHost::engineeringSettingsAllowed()
+{
+#if PLATFORM(COCOA)
+    static bool allowsInternalSecurityPolicies = os_variant_allows_internal_security_policies("com.apple.WebKit");
+
+    return allowsInternalSecurityPolicies && m_frontendPage->settings().webInspectorEngineeringSettingsAllowed();
+#else
+    return false;
+#endif
```

Comment:
NIT: I'd maybe reorganize this to avoid the `#else` and whatnot
```
    if (!m_frontendPage->settings().webInspectorEngineeringSettingsAllowed())
        return false;

#if PLATFORM(COCOA)
    static bool allowsInternalSecurityPolicies = os_variant_allows_internal_security_policies("com.apple.WebKit");
    if (!allowsInternalSecurityPolicies)
        return false;
#endif

    return true;
```
i.e. other platforms might also want the use of `WebInspectorEngineeringSettingsAllowed` :)

Reply:
Good call.